diff --git a/modprobe-utils/nvidia-modprobe-utils.c b/modprobe-utils/nvidia-modprobe-utils.c
--- a/modprobe-utils/nvidia-modprobe-utils.c
+++ b/modprobe-utils/nvidia-modprobe-utils.c
@@ -933,10 +933,10 @@
     return mknod_helper(major, minor_num, vgpu_dev_name, NV_PROC_REGISTRY_PATH);
 }

-static int nvidia_cap_get_device_file_attrs(const char* cap_file_path,
-                                            int *major,
-                                            int *minor,
-                                            char *name)
+int nvidia_cap_get_device_file_attrs(const char* cap_file_path,
+                                     int *major,
+                                     int *minor,
+                                     char *name)
 {
     char field[32];
     FILE *fp;
diff --git a/modprobe-utils/nvidia-modprobe-utils.h b/modprobe-utils/nvidia-modprobe-utils.h
--- a/modprobe-utils/nvidia-modprobe-utils.h
+++ b/modprobe-utils/nvidia-modprobe-utils.h
@@ -89,6 +89,7 @@
 int nvidia_nvswitch_get_file_state(int minor);
 int nvidia_cap_mknod(const char* cap_file_path, int *minor);
 int nvidia_cap_get_file_state(const char* cap_file_path);
+int nvidia_cap_get_device_file_attrs(const char* cap_file_path, int *major, int *minor, char *name);
 int nvidia_cap_imex_channel_mknod(int minor);
 int nvidia_cap_imex_channel_file_state(int minor);
 int nvidia_get_chardev_major(const char *name);
